lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -81,19 +107,45 @@ public function get_providers() {
 	public function register_font( $font ) {
 		$font = $this->validate_font( $font );
 		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			$this->webfonts[ $id ] = $font;
+			$slug = $this->get_font_slug( $font );
+
+			if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+				$this->registered_webfonts[ $slug ] = array();
+			}
+
+			$this->registered_webfonts[ $slug ][] = $font;
+		}
+	}
```

Comment:
A few suggestions:
- Type hint the method's parameter with `array` as `array` is the only expected data type.
- For consistency, use `$webfont` instead of `$font` as `webfont` refers to the configuration of a web font (i.e. refers to the list of arguments).
- For readability, convert the `if` conditional into a guard clause and bail out early if not valid. This shifts all the code remaining in the method to the left and clearly shows the intent that if not validate, it's done and no work will happen.

```php
	/**
	 * Register a webfont.
	 *
	 * @param array $webfont The web font's arguments.
	 */
	public function register_font( array $webfont ) {
		$webfont = $this->validate_font( $webfont );

		// If not valid, bail out.
		if ( empty( $webfont ) ) {
			return;
		}

		$slug = $this->get_font_slug( $webfont );

		// Initialize a new font-family collection.
		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
			$this->registered_webfonts[ $slug ] = array();
		}

		$this->registered_webfonts[ $slug ][] = $webfont;
	}
```